xdsl/irdl/attributes.py

```diff
@@ -172,15 +169,54 @@ def from_pyrdl(
             if key not in _PARAMETRIZED_ATTRIBUTE_DICT_KEYS
         }
 
-        # Check that all fields of the attribute definition are either already
-        # in ParametrizedAttribute, or are class functions or methods.
+        # The resulting parameters
+        parameters: dict[str, AttrConstraint] = {}
+
+        for field_name, field_type in get_type_hints(
```

Comment:
Can we move this logic in a separate function?
Just something that returns the parameters value.

Reply:
OK so I think this is more readable now. I'd really like to avoid helper functions, I think they just make things harder to read by making the reader go to definition and back, and now everything is in one place that is close to fitting on one screen. Once we merge this, we can get rid of the Annotated case, which will further shrink this code, let's iterate then?